### PR TITLE
Energy-device-related improvements

### PR DESCRIPTION
**Pull request by bouwew**
Summary:
- Correct typing for P1 sensors
- Remove `_energy_device_info_finder()` function, redistribute and improve the related code.
- Add guarding as suggested by coderabbitai

## Release Notes

- **New Features**
  - Enhanced data collection for appliances, excluding specific types for improved accuracy.
  - Added handling for a new measurement category, `SPECIALS`, improving measurement processing.

- **Bug Fixes**
  - Updated changelog to clarify fixes and improvements related to P1 sensors and energy devices.
  - Improved error handling for HTTP requests, particularly for timeout scenarios.

- **Documentation**
  - Updated changelog entries for clarity on function outputs and parameter changes.

- **Chores**
  - Project version updated from 1.5.0 to 1.5.1.

### CODE REVIEW

**Review by CoMPaTech — CHANGES_REQUESTED** [review]
The XML values depict floats, even if Plugwise doesn't decide to use them, we still should adhere to what is in the XML data?

File: plugwise/util.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def check_model(name: str | None, vendor_name: str | None) -> str | None:
     if name is not None and "lumi.plug" in name:
         return "Aqara Smart Plug"
 
-    return name
+    return name  # pragma: no cover
```
Comment (CoMPaTech) [inline plugwise/util.py:118]: We should check on this one, since it's the default it would be weird as being untestable?

Comment (bouwew) [reply plugwise/util.py:118]: This is the fallback-case, I could leave it out but then ruff will complain: missing return.

Comment (CoMPaTech) [reply plugwise/util.py:118]: Let's leave it be :)

**Review by CoMPaTech — APPROVED** [review]

File: plugwise/util.py (hunk 1 above)
Comment (bouwew) [reply plugwise/util.py]: This return can be removed apparently.

**Comment by bouwew** [comment]
The P1 sensor all-float-typing will be implemented in a new PR.

File: plugwise/util.py (hunk 1 above)
Comment (bouwew) [reply plugwise/util.py]: No it can't, mypy now complains :(

**Pull request merged**